### PR TITLE
Modify response header/body extraction method

### PR DESCRIPTION
**Pull request by thomas-klein**
Avoid bug happening when there is more than 1 header which actually cause $body to be the second header. 
We now assume that response body is ever the last part of the response (and not the second part).

### CODE REVIEW

**Comment by mglcel** [comment]
Hi Thomas, can you describe more precisely the initial problem you had please ?

Consider this script :
<?
$result = "X-Account-Object-Count: 4
X-Account-Bytes-Used: 3072798
X-Account-Container-Count: 3
Accept-Ranges: bytes
Content-Length: 15
Content-Type: text/plain; charset=utf-8
Date: Thu, 12 Jan 2012 04:07:33 GMT\r
\r
abc
testing";

print "*\* Impl A";
list($headers, $body) = explode("\r\n\r\n", $result);
print "\n== HEADERS: == \n" . print_r($headers, true);
print "\n== BODY: == \n" . print_r($body, true);

print "\n*\* Impl B";
$result_sections = explode("\r\n\r\n", $result);
$body = end($result_sections);
$header = prev($result_sections);
print "\n== HEADERS: == \n" . print_r($header, true);
print "\n== BODY: == \n" . print_r($body, true);
?>

And the output i have :
*\* Impl A
== HEADERS: ==
X-Account-Object-Count: 4
X-Account-Bytes-Used: 3072798
X-Account-Container-Count: 3
Accept-Ranges: bytes
Content-Length: 15
Content-Type: text/plain; charset=utf-8
Date: Thu, 12 Jan 2012 04:07:33 GMT
== BODY: ==
a1
abc
testing

*\* Impl B
== HEADERS: ==
X-Account-Object-Count: 4
X-Account-Bytes-Used: 3072798
X-Account-Container-Count: 3
Accept-Ranges: bytes
Content-Length: 15
Content-Type: text/plain; charset=utf-8
Date: Thu, 12 Jan 2012 04:07:33 GMT
== BODY: ==
a1
abc
testing

It seems the result is the same, it's why i need more details about your initial issue..

Thanks !
++

**Comment by thomas-klein** [comment]
I got the following problem :

``` php
$VD->post('/joboffer')->offers($xml)->x(); // return null
```

After digging into the PHP SDK code and using some var_dump/echo I find out that the $body of the response contained : "HTTP 400 Bad Request..." which passed to json_decode gave me the "null". And I find that the $header contained "HTTP 100 Continue". 
So I conclude that you can't reliably use the second part of the result as the body and better use the last part of it instead.

To summarize I had:

``` php
$header == "HTTP 100 Continue"
$body == "HTTP 400 Bad Request..."
```

And the real body which was in this case the description of the error was lost and never returned...

I guess the response was something like this:

> HTTP 100 Continue\r
> \r
> HTTP 400 Bad Request\r
> \r
> {json_describing_the_error}

I hope it's a bit clearer and I didn't lost you in my attempt at english speaking which is not my best skill (I'm french).

**Comment by mglcel** [comment]
ok, got it, it's specific to the HTTP/1.1 Continue mechanism.
Your solution is the correct one.
Thanks !

**Comment by thomas-klein** [comment]
Glad I could help.

Also it's the first pull request I ever made and it has been accepted so thank you ! ;)

**Comment by mglcel** [comment]
you're welcome ! :)
